In [ ]:
import zipfile
import os
from PIL import Image
import shutil

In [ ]:
cleanadv = 'clean_and_adv.zip'
cleanadv_ref = zipfile.ZipFile(cleanadv, 'r')
cleanadv_ref.extractall()

In [ ]:
cleanadv_ref.close()

In [ ]:
# 1. new directory
# move every fifth file to test

In [ ]:
parent = "/content/"
new_dir = "attacktest"

path = os.path.join(parent, new_dir)
os.mkdir(path)

In [ ]:
f = 1
for filename in os.scandir('attacks'):
  if filename.is_file():
    if f % 5 == 0:
      shutil.move(filename, 'attacktest')
      print(str(filename) + "to test set")
    f += 1
    print(filename)

<DirEntry 'img358.png'>
<DirEntry 'img551.png'>
<DirEntry 'img440.png'>
<DirEntry 'img224.png'>
<DirEntry 'img690.png'>to test set
<DirEntry 'img690.png'>
<DirEntry 'img561.png'>
<DirEntry 'img339.png'>
<DirEntry 'img233.png'>
<DirEntry 'img305.png'>
<DirEntry 'img567.png'>to test set
<DirEntry 'img567.png'>
<DirEntry 'img715.png'>
<DirEntry 'img722.png'>
<DirEntry 'img130.png'>
<DirEntry 'img137.png'>
<DirEntry 'img259.png'>to test set
<DirEntry 'img259.png'>
<DirEntry 'img503.png'>
<DirEntry 'img688.png'>
<DirEntry 'img619.png'>
<DirEntry 'img8.png'>
<DirEntry 'img290.png'>to test set
<DirEntry 'img290.png'>
<DirEntry 'img134.png'>
<DirEntry 'img517.png'>
<DirEntry 'img327.png'>
<DirEntry 'img648.png'>
<DirEntry 'img499.png'>to test set
<DirEntry 'img499.png'>
<DirEntry 'img240.png'>
<DirEntry 'img258.png'>
<DirEntry 'img659.png'>
<DirEntry 'img445.png'>
<DirEntry 'img61.png'>to test set
<DirEntry 'img61.png'>
<DirEntry 'img148.png'>
<DirEntry 'img656.png'>
<DirEntry 'img4.png'>
<Dir

In [ ]:
c=1
for filename in os.scandir('attacktest'):
    if filename.is_file():
      c += 1

print(c)

147


In [ ]:
c=1
for filename in os.scandir('attacks'):
    if filename.is_file():
      c += 1

print(c)

589


Attacked images:

589 train samples

147 test samples

80/20 train test split

In [ ]:
# do the same for clean images

In [ ]:
parent = "/content/"
new_dir = "cleantest"

path = os.path.join(parent, new_dir)
os.mkdir(path)

In [ ]:
f = 1
for filename in os.scandir('clean'):
  if filename.is_file():
    if f % 5 == 0:
      shutil.move(filename, 'cleantest')
      print(str(filename) + "to test set")
    f += 1
    print(filename)

<DirEntry 'road416.png'>
<DirEntry 'road368.png'>
<DirEntry 'road798.png'>
<DirEntry 'road758.png'>
<DirEntry 'road752.png'>to test set
<DirEntry 'road752.png'>
<DirEntry 'road849.png'>
<DirEntry 'road71.png'>
<DirEntry 'road77.png'>
<DirEntry 'road271.png'>
<DirEntry 'road206.png'>to test set
<DirEntry 'road206.png'>
<DirEntry 'road349.png'>
<DirEntry 'road774.png'>
<DirEntry 'road235.png'>
<DirEntry 'road519.png'>
<DirEntry 'road212.png'>to test set
<DirEntry 'road212.png'>
<DirEntry 'road296.png'>
<DirEntry 'road740.png'>
<DirEntry 'road535.png'>
<DirEntry 'road91.png'>
<DirEntry 'road320.png'>to test set
<DirEntry 'road320.png'>
<DirEntry 'road297.png'>
<DirEntry 'road162.png'>
<DirEntry 'road471.png'>
<DirEntry 'road342.png'>
<DirEntry 'road522.png'>to test set
<DirEntry 'road522.png'>
<DirEntry 'road220.png'>
<DirEntry 'road6.png'>
<DirEntry 'road495.png'>
<DirEntry 'road687.png'>
<DirEntry 'road462.png'>to test set
<DirEntry 'road462.png'>
<DirEntry 'road592.png'>
<DirEntry 'roa

In [ ]:
c=1
for filename in os.scandir('cleantest'):
    if filename.is_file():
      c += 1

print(c)

176


In [ ]:
c=1
for filename in os.scandir('clean'):
    if filename.is_file():
      c += 1

print(c)

703


Clean images:

176 test samples

703 train samples

In [ ]:
# 1. rename folders
# 2. create new main dataset folder
# 3. transfer all folders
# 4. save to .zip

In [ ]:
os.mkdir('/content/clean_adversarial')

In [ ]:
os.mkdir('/content/attack')
os.mkdir('/content/clean')

In [ ]:
shutil.move('attacktrain', 'attack/attacktrain')
shutil.move('attacktest', 'attack/attacktest')

shutil.move('cleantrain', 'clean/cleantrain')
shutil.move('cleantest', 'clean/cleantest')

'clean/cleantest'

In [ ]:
shutil.move('attack', 'clean_adversarial/attack')
shutil.move('clean', 'clean_adversarial/clean')

'clean_adversarial/clean'

In [ ]:
shutil.make_archive('clean_adversarial', 'zip', 'clean_adversarial')

'/content/clean_adversarial.zip'